pkg/controllers/rollout/controller.go

```diff
@@ -407,7 +407,7 @@ func (r *Reconciler) pickBindingsToRoll(ctx context.Context, allBindings []*flee
 			if bindingutils.HasBindingFailed(binding) {
 				klog.V(3).InfoS("Found a failed to be ready bound binding", "clusterResourcePlacement", crpKObj, "binding", bindingKObj)
 				bindingFailed = true
-			} else {
+			} else if !bindingutils.IsBindingReportDiff(binding) {
```

Comment:
A similar pattern happens here as well:

Since `HasBindingFailed` will only return `True` when either `Applied` or `Available` condition is set to False and fresh, any binding in `ReportDiff` mode will never be considered to be failing.

Reply:
So `ReportDiff` bindings are always updated per the `maxUnavailable` setting, even if the `DiffReported` condition has turned False (an error has occurred).

Reply:
Not to say that it is a wrong behavior though. Again, it is just an inconsistency.